### ISSUE TITLE
Storage from @ionic/storage loses data

### ISSUE THREAD

**Issue by jgw96** [driftyco/ionic-storage#39]
_From @Lafaa on October 30, 2016 16:29_

Hi guys!
Am i the only one having problem with the new Storage since the ionic2 rc0? 

It just forgets everything once the app is restarted. I use the Storage class from @ionic/storage to save username and password, but it's like it get cleaned on app restart. 

I'm using the Storage this way:

```
 public storage: Storage;
...
this.storage = new Storage();
...
this.storage.set(key, value);

```

I've also added the Storage in app.module.ts, both in the import section AND in the providers array. 
Did i get something wrong? In the running and building process i don't have any error or warnings

Cordova CLI: 6.3.1
Ionic Framework Version: 2.0.0-rc.0-201610131811
Ionic CLI Version: 2.1.0
Ionic App Lib Version: 2.1.0-beta.1
OS:
Node Version: v6.7.0

_Copied from original issue: driftyco/ionic#8959_

**Comment by jgw96** [comment]
Hello, thanks for using Ionic! It seems that this issue is most likely caused by your use of `new` with storage and by putting `storage` in your `imports` array. Could you take a look at the docs for [ionic storage](https://github.com/driftyco/ionic-storage) and give it another try? Thanks!

**Comment by jgw96** [comment]
_From @jherediagu on November 1, 2016 11:27_

Yes, it's lost data, I have the same problem, I solve it with localforage, like this:

```
import localforage from 'localforage';

   localforage.setItem<any>('foo', 'bar');

   localforage.getItem<any>('foo').then( (value:any) => {
      console.log(value);
   });
```

**Comment by jgw96** [comment]
Thanks for the info. Unfortunately I cannot reproduce this issue, would you posting a repo or a zip of your project that I can use to reproduce this issue?

**Comment by tabirkeland** [comment]
I was seeing similar behavior and then wrapped my storage method calls in ```platform.ready().then(() => { });```.

Solved my problems.

**Comment by mirkonasato** [comment]
Could be the same issue as #20

**Comment by FdezRomero** [comment]
@jgw96 I'm having the same problem on Android with the latest versions, whereas 1.0.x were working okay. I checked just in case and I'm not using `new` nor have Storage in my app module imports.

I agree that it seems related to #20, as I noticed to have keys in `localstorage` and Storage should always pick SQLite as the engine (I have the plugin installed).

Thanks!

**Comment by mirkonasato** [comment]
I submitted [PR 19](https://github.com/driftyco/ionic-storage/issues/19) weeks ago as a possible fix for these issues. Would be nice if somebody could look at it.

**Comment by paulovitorjp** [comment]
I am wrapping everything inside platform.ready() as you can see below, and the get returns null everytime.

```
  constructor(platform: Platform, public deploy: Deploy,public modalCtrl: ModalController, public authSrvc: AuthService,
    public localStorage:Storage, public chatSrvc:ChatService, public prefSrvc:PrefService) {
    platform.ready().then(() => {
      // Okay, so the platform is ready and our plugins are available.
      // Here you can do any higher level native things you might need.
      StatusBar.styleDefault();
      console.log('[App Component] SQLite length: ');
      localStorage.length().then((n)=>{console.log('[App Component] SQLite length: ' + n);});
      localStorage.set('test',true).then(()=>{
        localStorage.get('test').then((test)=>{
          console.log('testing of sqlite was ' + test); //this is always null, even though I just set it to true.
          //...
        });
      });
    });
  }
```

I've seen that @mirkonasato has submitted a pull request for this here #21 . Please @jgw96, could you take a look?

I am using ionic 2.1.8, cordova 6.3.1, ionic storage 1.1.6 and cordova-sqlite-storage 1.5.0.

Thanks!

**Comment by abhishekpal123** [comment]
I faced the same issue and found a workaround. I initialised the rootPage inside platform.ready(). The code looks like this:
--------------
export class App {
. . .
rootPage;

  constructor(private storage: Storage, public platform: Platform) {
      this.platform.ready().then(() => {
        this.rootPage = <your login page>;
        StatusBar.styleDefault();
        Splashscreen.hide();
      });
  }
. . .

--------------
This works for me. Do try and share your experience or thoughts.

**Comment by mlynch** [comment]
Fixed with #21
